### DISCUSSION TITLE
Getting delayed pose data with Lighthouse Positioning System

### DISCUSSION THREAD

**Discussion by peihan-li** [USC-ACTLab/crazyswarm#774] [Q&A] [answered]
Hello @whoenig, I'm doing research in multi-agent target tracking and using the crazyflie drones with the Crazyswarm framework to control the drones tracking targets. For the positioning system, we are using a LightHouse system with 4 base stations in our lab. In my code, I will need to get the drone's location to plan the next command. However, I found using `cf.position()` command doesn't have enough update rate (~2 Hz), and using logging information `/cf/log1` published on ROS is lagging around 1-2 seconds.

I tested this by holding the drone and move it around and looking at the values printout from `/cf/log1`, I can see when I move the drone, the position data doesn't change immediately, but would have a ~2 sec delay. 

I'm wondering if this is a known issue and how do I get real-time position reading using in the python code.

Thank you! Please let me know if you need more information.

**Comment by whoenig** [comment, marked as answer]
This is a known problem in Crazyswarm1 - it works fine with motion capture but not with LH/LPS if you need the position on the host side. You can try increasing the ping frequency at https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/src/crazyswarm_server.cpp#L971 (change the 10 to 1), which can help to reduce the delay.

In Crazyswarm2 we completely rewrote the communication backend and the latest version has a built-in latency measurement and one gets delays of <10ms for one drone (haven't tested it much with 4 and LH, but should also be similar).

**Reply by peihan-li** [reply]
Thank you for the help! The latency decreased a lot after changing the 10 to 1.
